powerplantmatching/osm/enhancement/reconstruction.py

```diff
@@ -0,0 +1,348 @@
+"""Plant reconstruction from incomplete OSM data.
+
+This module provides functionality to reconstruct power plants from
+orphaned generators and incomplete plant relations.
+"""
+
+import logging
+import re
+from typing import TYPE_CHECKING, Any, Optional
+
+if TYPE_CHECKING:
+    from powerplantmatching.osm.parsing.generators import GeneratorParser
+
+logger = logging.getLogger(__name__)
+
+
+class NameAggregator:
+    """Aggregates multiple generator names into a plant name.
+
+    Uses n-gram analysis to find common substrings among generator
+    names to create a meaningful plant name.
+
+    Attributes
+    ----------
+    config : dict
+        Configuration settings
+    similarity_threshold : float
+        Minimum similarity ratio for common substrings
+    """
+
+    def __init__(self, config: dict[str, Any]):
```

Comment:
This class only needs `name_similarity_threshold` - why give it the whole config?